tests/unit/LSYiiApplicationTest.php

```diff
@@ -0,0 +1,178 @@
+
+        $this->assertSame($url, 'http://absoluteConfig.example.com/', 'Unexpected url. The url does not correspond to the one previously set.');
+
+        // Reset original value.
+        Yii::app()->setConfig('publicurl', $tmpPublicUrl);
+    }
+
+    /**
+     * Get public base url, previously set in baseUrl request attribute.
+     */
+    public function testGetpublicBaseUrlFromRequest()
+    {
+        $tmpPublicUrl = Yii::app()->getRequest()->getBaseUrl();
+
+        Yii::app()->getRequest()->baseUrl = 'http://request.example.com/';
+        $url = Yii::app()->getPublicBaseUrl();
+
+        $this->assertSame($url, 'http://request.example.com/', 'Unexpected url. The url does not correspond to the one previously set.');
+
+        // Reset original value.
+        Yii::app()->getRequest()->baseUrl = $tmpPublicUrl;
+    }
+
+    /**
+     * Get absolute public base url, previously set in baseUrl request attribute.
+     */
+    public function testGetAbsolutepublicBaseUrlFromRequest()
+    {
+        $tmpPublicUrl = Yii::app()->getRequest()->getBaseUrl();
+
+        Yii::app()->getRequest()->baseUrl = 'absoluteRequest.example.com/';
+        $url = Yii::app()->getPublicBaseUrl(true);
+
+        $this->assertSame($url, 'http://absoluteRequest.example.com/', 'Unexpected url. The url does not correspond to the one previously set.');
+
+        // Reset original value.
+        Yii::app()->getRequest()->baseUrl = $tmpPublicUrl;
+    }
+
+    /**
+     * Get public base url, previously set in baseUrl request attribute.
+     * A public url is also set in the baseUrl config attribute,
+     * but the one in the request attribute should be returned.
+     */
+    public function testGetpublicBaseUrlFromConfigNoSchemeInConfigPublicUrl()
+    {
+        $tmpConfigPublicUrl = Yii::app()->getConfig('publicurl');
+        $tmpRequestPublicUrl = Yii::app()->getRequest()->getBaseUrl();
+
+        // No scheme in url.
+        Yii::app()->setConfig('publicurl', '//config.example.com/path?param=1');
+        Yii::app()->getRequest()->baseUrl = 'http://request.example.com/';
+        $url = Yii::app()->getPublicBaseUrl();
+
+        $this->assertSame($url, 'http://request.example.com/', 'Unexpected url. The url does not correspond to the one previously set.');
+
+        // Restore original values.
+        Yii::app()->setConfig('publicurl', $tmpConfigPublicUrl);
+        Yii::app()->getRequest()->baseUrl = $tmpRequestPublicUrl;
+    }
+
+    /**
+     * Get public base url, previously set in baseUrl request attribute.
+     * A public url is also set in the baseUrl config attribute,
+     * but the one in the request attribute should be returned.
+     */
+    public function testGetpublicBaseUrlFromConfigNoHostInConfigPublicUrl()
+    {
+        $tmpConfigPublicUrl = Yii::app()->getConfig('publicurl');
+        $tmpRequestPublicUrl = Yii::app()->getRequest()->getBaseUrl();
+
+        Yii::app()->setConfig('publicurl', 'http://');
```

Comment:
My opinion : test must break here : publicurl must be always used at last option, even if it's a broken url